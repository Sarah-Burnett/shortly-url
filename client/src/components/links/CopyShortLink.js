import React from 'react';
import styled from 'styled-components';
import { breakpoint, cyan } from '../../Styles';
import { shortlyUrl } from '../../utilities/url';

const ShortUrl = styled.input`
    flex: 2;
    color: ${cyan};
    display: block;
    border: none;
    width: 100%;
    height: 5vh;
    margin: 2vh 0 2vh 0;
    outline: none;
    font-size: inherit;
    font-family: inherit;
    @media (min-width: ${breakpoint}) {
        text-align: right;
        margin: 0 2vw 0 2vw;
        width: auto;
    }
    overflow: hidden;
`

export const CopyShortLink = ({shortUrl}) => {
    return (
        <ShortUrl readOnly="readonly" className="shortLink" value={shortlyUrl + shortUrl}/>
    )
}