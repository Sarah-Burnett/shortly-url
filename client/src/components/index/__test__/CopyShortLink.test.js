import React from 'react';
import {
	rendersWithoutCrashing,
	matchesSnapshot,
} from "../../../setupTests";
import CopyShortLink from "../CopyShortLink";

const CopyShortLinkWrapper = () => <CopyShortLink shortUrl="shortUrl"/>

it("renders copy short link without crashing", () =>
	rendersWithoutCrashing(CopyShortLinkWrapper));

it("copy short link matches snapshot", () => matchesSnapshot(CopyShortLinkWrapper));
