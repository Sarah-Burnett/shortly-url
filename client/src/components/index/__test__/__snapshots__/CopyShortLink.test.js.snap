// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`copy short link matches snapshot 1`] = `
<input
  className="sc-fzplWN bbpMZr shortLink"
  readOnly="readonly"
  value="https://short.ly/shortUrl"
/>
`;
